python/lsst/skymap/baseSkyMap.py

```diff
@@ -160,3 +164,49 @@ def __iter__(self):
 
     def __len__(self):
         return len(self._tractInfoList)
+
+    def getSha1(self):
+        """Return a SHA1 hash that uniquely identifies this SkyMap instance.
+
+        Returns
+        -------
+        sha1 : bytes
+            A 20-byte hash that uniquely identifies this SkyMap instance.
+
+        Subclasses should almost always override `updateSha1()` instead of
+        this function to add subclass-specific state to the hash.
+        """
+        sha1 = hashlib.sha1()
+        # Note that str() calls below are needed to coerce to unicode in
+        # Python 2 (note that this `str` is from `future`), before
+        # we coerce back to bytes to be compatible with Python 3.
+        # In a pure Python 3 world, we can remove the str() calls.
+        sha1.update(bytes(str(type(self).__name__), encoding='utf-8'))
+        configPacked = struct.pack(
+            "<iiidd3sd",
+            self.config.patchInnerDimensions[0],
+            self.config.patchInnerDimensions[1],
+            self.config.patchBorder,
+            self.config.tractOverlap,
+            self.config.pixelScale,
+            bytes(str(self.config.projection), encoding='latin1'),
```

Comment:
why latin1 here?

Reply:
These strings are required by the config to be one of a few 3-letter FITS WCS projection names, so I know `latin1` is sufficient for representing - until FITS supports unicode in headers, that is :-)

That said, I don't actually know whether knowing it's sufficient means I should use it over something else.  I suspect it just doesn't matter, but I'd love to get any feedback you have on that.

Reply:
I would expect ascii for a projection code.

Reply:
It turns out this doesn't work anyway, because `str()` doesn't take an `encoding` argument in Python 2.  I've switched to just using `future.utils.tobytes` instead, which I think just uses `latin1` under the hood.